### ISSUE TITLE
MSD slow (data loading slow?)

### ISSUE THREAD

**Issue by christophlohrmann** [zincware/MDSuite#370]
Calculating a MSD for ~100Mb trajectory takes several minutes.

**Comment by SamTov** [comment]
What are the parameters?

**Comment by christophlohrmann** [comment]
atom_selection = all atoms (100)
tau_values = 500 logspaced points in [0,10'000] from a trajectory that is 100'000 samples long

**Comment by SamTov** [comment]
Correlation time is 1 then? Because in that case you have 90'000 msds being computed. In general that also should not take too long but I am unsure of the slow down incurred by the tau value slicing. How many minutes are we talking?

**Comment by christophlohrmann** [comment]
yes correlation time is 1. Are the ensembles batched one by one (i.e. every single ensemble calculation loads from the database)? Because the whole thing easily fits in memory so it could be loaded all together.
We are talking between 15 and 45 minutes. not sure why it has such a large variance

**Comment by SamTov** [comment]
That is quite odd. ensembles should all be sitting in memory. What is the loop number of the tqdm bar? If 1, then it is all in memory.

**Comment by christophlohrmann** [comment]
it is a 1-loop, yes

**Comment by SamTov** [comment]
Okay that is interesting then. I can't reproduce this on my largest datasets. I would need to see where it is spending all of its time as it isn't in the data loading if there is only a single batch.

**Comment by christophlohrmann** [comment]
for now I can live with the slowdown. Maybe it is because there are a lot of msds for the small tau values

**Comment by SamTov** [comment]
I could imagine that it is due to many MSDs but you don't have many particles right? I think we need to check whether many computations slows it down more than bigger computations or whether we can run some ensemble loops in parallel if a system only has a few atoms. It would be unnecessary to use 10 cores for 10 particles when we could use those cores to run ensembles in parallel.
